cirq/ops/common_gates.py

```diff
@@ -116,6 +116,30 @@ def _trace_distance_bound_(self) -> Optional[float]:
             return None
         return abs(np.sin(self._exponent * 0.5 * np.pi))
 
+    def controlled(self,
+                   num_controls: int = None,
+                   control_values: Optional[Sequence[
+                       Union[int, Collection[int]]]] = None,
+                   control_qid_shape: Optional[Tuple[int, ...]] = None
+                  ) -> raw_types.Gate:
+        """
+        Specialize controlled method for XPowGate to construct a corresponding
+        controlled CNotPowGate, if applicable.
```

Comment:
I still think this pydoc could use some improvement.  Typically pydoc is a one-line summary with a more in-depth paragraph form to follow.

Here is an example of what I am thinking:

```
Constructs CNotPowGate from controlled XPowGate when applicable.

This method is a specialized controlled method for XPowGate.  It overrides
the default behavior of returning a ControlledGate by transforming the
underlying controlled gate to a CNotGatePow and removing the last
specified control qubit (which acts first semantically).  If this is a gate with
multiple control qubits, it will now be a ControlledGate with one less control.

This behavior only occurs when the last control qubit is a default-type control qubit.

(Note that a CNotPowGate is, by definition, a controlled-X gate.)
```

Reply:
This definitely looks much better. Using this with some changes like adding the explanation for default-qubit below.